src/functionalTest/java/uk/gov/hmcts/reform/sendletter/util/LetterStatusDeserialiser.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.reform.sendletter.util;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import uk.gov.hmcts.reform.sendletter.domain.LetterStatus;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+public class LetterStatusDeserialiser extends StdDeserializer<LetterStatus> {
```

Comment:
I am not sure why we will need a deserialiser for parsing and asserting the response object.You can just use `@JsonCreator` annotation to tell Jackson which constructor to use while deserialising object.
```
@JsonCreator
    public LetterStatus(
        @JsonProperty("id") final UUID id,
        @JsonProperty("message_id") final String messageId,
        @JsonProperty("created_at") final ZonedDateTime createdAt,
        @JsonProperty("sent_to_print_at") final ZonedDateTime sentToPrintAt,
        @JsonProperty("printed_at") final ZonedDateTime printedAt,
        @JsonProperty("has_failed") final boolean hasFailed
    )
```
Also you will need below config in `application.properties`
```spring.jackson.deserialization.adjust_dates_to_context_time_zone=false```

I have tested this on my local and it works.Feel free to add if we have a better approach.

Reply:
had this before but assumed wrongly that it is not preferred since letter is not ever used to deserialise only in test. can simply revert that commit really.

yep, have that config set ;)

Reply:
You can simply test the json instead of deserializing the response string into letter status object and then asserting.But if you want to deserialize then your best bet is to go with above approach :)

Reply:
wanted to deserialise, for some reason - don't ask 😄. i'll just do a raw json comparison then

Reply:
Can we just make asserts on json?
Spring includes [JSONassert](https://github.com/skyscreamer/JSONassert) and [JsonPath](https://github.com/json-path/JsonPath)